pkg/status/status.go

```diff
@@ -28,8 +28,52 @@ type status struct {
 
 // Status is the interface to report the health of the operator to CVO
 type Status interface {
-	SetFailing(message string)
-	SetAvailable(message string)
+	SetStatus(conditions []Condition)
+}
+
+// Condition is a struct used to configure operator status conditions
+type Condition struct {
+	conditionType   configv1.ClusterStatusConditionType
+	conditionStatus configv1.ConditionStatus
+	message         string
+}
+
+// NewAvailableCondition return an available condition based on the status and message provided
+func NewAvailableCondition(condititionStatus bool, message string) Condition {
+	if condititionStatus {
+		return createCondition(configv1.OperatorAvailable, configv1.ConditionTrue, message)
+	} else {
+		return createCondition(configv1.OperatorAvailable, configv1.ConditionFalse, message)
+	}
+}
+
+// NewProgressingCondition return a progressing condition based on the status and message provided
+func NewProgressingCondition(condititionStatus bool, message string) Condition {
+	if condititionStatus {
+		return createCondition(configv1.OperatorProgressing, configv1.ConditionTrue, message)
+	} else {
+		return createCondition(configv1.OperatorProgressing, configv1.ConditionFalse, message)
+	}
+}
+
+// NewFailingCondition return a failing condition based on the status and message provided
+func NewFailingCondition(condititionStatus bool, message string) Condition {
+	if condititionStatus {
+		return createCondition(configv1.OperatorFailing, configv1.ConditionTrue, message)
+	} else {
+		return createCondition(configv1.OperatorFailing, configv1.ConditionFalse, message)
+	}
+}
```

Comment:
I don't foresee a difference outside of the `configv1.ConditionStatus` parameter. I had avoided converging these methods into a single method because I wanted the calls to be clear, concise, and not require `configv1` to be imported. I do see the benefit of generalizing the method, we could delete the `NewAvailableCondition/NewProgressingCondition/NewFailingCondition` methods and update existing calls to reference the `createCondition`  method.

Reply:
I agree with @kevinrizza. This seems like a lot of repeated code. Not sure how much we are gaining here vs the initial implementation which was simpler.

Reply:
Understood, I will remove the `NewAvailableCondition/NewProgressingCondition/NewFailingCondition` methods

> Not sure how much we are gaining here vs the initial implementation which was simpler.

While the original implementation was simpler, we could not update individual condition statuses while leave others conditions as they were .

Reply:
>While the original implementation was simpler, we could not update individual condition statuses while leave others conditions as they were .
Isn't there a risk of them being changed in an non-atomic manner? Moreover do we have that use case.

Reply:
> Isn't there a risk of them being changed in an non-atomic manner? 

The only method that changes the status is `status.UpdateStatus` which is called after updating all the conditions - I don't see a point where these statuses could be changed individually..

> Moreover do we have that use case?

Marketplace doesn't take advantage of setting individual conditions yet so I could move this into a method similar to the one we had earlier if you don't see this being a usecase in the future.

Reply:
>Marketplace doesn't take advantage of setting individual conditions yet so I could move this into a method similar to the one we had earlier if you don't see this being a usecase in the future.


I prefer introducing this level of changes when it is actually needed. Plus we are close to GA, so less code changes the better.

Reply:
Unerstood, I will update the code accordingly.